compiler/rustc_codegen_llvm/src/llvm/mod.rs

```diff
@@ -1,4 +1,6 @@
 #![allow(non_snake_case)]
+#![expect(dead_code)]
+#![expect(unreachable_pub)]
```

Comment:
I'm reluctant to do this, even temporarily, because it has the side-effect of disabling dead-code warnings throughout the whole module, even for things that *should* be warnings (e.g. an unused `pub(crate)` helper function).

And we would be unable to get rid of it until *all* of the unwanted warnings are fixed individually.

Reply:
hmm. how do nested `deny` and `expect` work, exactly?

Reply:
IIRC, `expect` behaves exactly like `allow`, except that it also emits an `unfulfilled_lint_expectations` (which is a warning by default) if it doesn't suppress anything.

So if you have warn/deny on an outer module, and then allow/expect on an inner module/item, the diagnostic is suppressed as you would expect.

Reply:
If you mean nested in the other direction, I have no idea.

(But I have noticed that if I put `#[allow(dead_code)]` on an enum, and then `#[warn(dead_code)]` on one of its variants, there is no diagnostic even if that variant is unused. And if I replace the `warn` with `expect`, the expectation is violated.)

Reply:
Ok, removed this and re-added the workaround in `lib.rs`